### ISSUE TITLE
Fix a loss function of a discriminator

### ISSUE THREAD

**Issue by notani** [MysteryVaibhav/vanGAN#8]
https://github.com/MysteryVaibhav/vanGAN/blob/bbb831a1a99a66fb1b181ea293472a0a4200a015/src/trainer.py#L89-L90

This should be `torch.ones` (or its smoothed version) because we want a discriminator to output high `d_real_decision` for `d_real_data`.

https://github.com/MysteryVaibhav/vanGAN/blob/bbb831a1a99a66fb1b181ea293472a0a4200a015/src/trainer.py#L100

This should be `torch.zeros` or its smoothed version.

Also, we should use `torch.zeros(batch_size)` instead of `torch.zeros(batch_size, 1)` as `d_real_decision` is already flattend. Same for `torch.zeros`

**Comment by MysteryVaibhav** [comment]
The current trainer.py seems to be a very old version.

**Comment by notani** [comment]
I noticed it, too. Will you update it and close issues if they're already fixed? Thanks!

**Comment by MysteryVaibhav** [comment]
Updated
